lib/connection/connection_config.js

```diff
@@ -145,13 +145,21 @@ function ConnectionConfig(options, validateCredentials, qaMode, clientInfo)
   // only validate credentials if necessary
   if (validateCredentials)
   {
-    // check for missing username
-    Errors.checkArgumentExists(Util.exists(options.username),
-      ErrorCodes.ERR_CONN_CREATE_MISSING_USERNAME);
+    // username is not required for oauth authenticator
+    if (!Util.exists(options.authenticator) ||
+      options.authenticator !== authenticationTypes.OAUTH_AUTHENTICATOR) 
+    {
+      // check for missing username
+      Errors.checkArgumentExists(Util.exists(options.username),
+        ErrorCodes.ERR_CONN_CREATE_MISSING_USERNAME);
+    }
 
-    // check for invalid username
-    Errors.checkArgumentValid(Util.isString(options.username),
-      ErrorCodes.ERR_CONN_CREATE_INVALID_USERNAME);
+    if (Util.exists(options.username))
+    {
+      // check for invalid username
+      Errors.checkArgumentValid(Util.isString(options.username),
```

Comment:
can we add this check in the newly added if above? the `if` body above checks user name existence and we don't need additional `if`. Additionally, in the same way it's done for password check below only for default authenticator

Reply:
That's actually my initial fix. But as Ilesh mentioned, we should also validate username for OAUTH if username exist.